crates/amaru-kernel/src/memoized.rs

```diff
@@ -0,0 +1,265 @@
+use pallas_addresses::Address;
+use pallas_codec::{minicbor as cbor, minicbor::data::IanaTag, utils::Bytes};
+use pallas_crypto::hash::{Hash, Hasher};
+use pallas_primitives::{
+    alonzo,
+    conway::{
+        KeepRaw, MintedDatumOption, MintedTransactionOutput, NativeScript, PseudoScript, Value,
+    },
+    DatumHash, NonEmptyKeyValuePairs, PlutusData,
+};
+
+// --------------------------------------------------- MemoizedTransactionOutput
+
+#[derive(Debug, Clone)]
+pub struct MemoizedTransactionOutput {
+    is_legacy: bool,
+    pub address: Address,
+    pub value: Value,
+    pub datum: MemoizedDatum,
+    pub script: Option<MemoizedScript>,
+}
+
+impl<'b, C> cbor::Decode<'b, C> for MemoizedTransactionOutput {
+    // TODO: This implementation still copies the bytes when we convert from the minted version to
```

Comment:
Yes, this is a great first step and makes things easier to work with, but the true benefit will be once we update the decoding.

Are we concerned that with Pallas v1 this "issue" will be solved, thus rendering this work obsolete? Or is the plan that this becomes the solution in Pallas v1